### PR TITLE
feat: add diffs group

### PR DESCRIPTION
**Pull request by jsfez**

### CODE REVIEW

**Review by gregberge — CHANGES_REQUESTED** [review]
You can miss one at this step, you should do it at the end of the diff.

File: apps/screenshot-diff/src/computeScreenshotDiff.ts [hunk 1]
```diff
@@ -67,20 +67,41 @@ async function ensureFileDimensionsAndPreload({
...
+  buildId: string;
+}) {
+  const file = await File.query().findOne({ key });
+  if (!file) return null;
+
+  await ScreenshotDiff.query()
+    .where({ buildId, s3Id: key, group: null })
+    .patch({ group: key });
+  return file;
+}
```
Comment (gregberge) [inline apps/screenshot-diff/src/computeScreenshotDiff.ts:87]: This is not needed.

File: apps/screenshot-diff/src/computeScreenshotDiff.ts [hunk 2]
```diff
@@ -229,14 +260,28 @@ export const computeScreenshotDiff = async (
...
+    .where({ buildId })
+    .whereNotNull("group");
+
+  await Promise.all(
+    diffsGroups.map(async (diffsGroup) =>
+      ScreenshotDiff.query()
+        .where({ buildId, s3Id: diffsGroup.s3Id, group: null })
+        .patch({ group: diffsGroup.group }),
+    ),
+  );
```
Comment (gregberge) [inline apps/screenshot-diff/src/computeScreenshotDiff.ts:275]:
```ts
ScreenshotDiff.query()
        .where({ buildId, s3Id: screenshotDiff.s3Id, group: null })
        .patch({ group: diffsGroup.group })
```

**Review by gregberge — CHANGES_REQUESTED** [review]
Remove all code except the patch at the end of the compute, it is enough.

File: apps/screenshot-diff/src/computeScreenshotDiff.ts (hunk 2 above)
Comment (gregberge) [inline apps/screenshot-diff/src/computeScreenshotDiff.ts:275]: We don't want to patch all groups, just the group concerning our screenshot diff.

```ts
const similarDiffCount = await ScreenshotDiff.query().where({ buildId, s3Id: screenshotDiff.s3Id }).resultSize()
if (similarDiffCount > 1) {
  await ScreenshotDiff.query().where({ buildId, s3Id: screenshotDiff.s3Id, group: null }).patch({ group: screenshotDiff.s3Id });
}
```

**Review by gregberge — CHANGES_REQUESTED** [review]

**Comment by jsfez** [comment]
I agree. You version is easier to maintain

File: apps/database/db/structure.sql [hunk 3]
```diff
@@ -401,7 +401,7 @@ ALTER SEQUENCE public.github_installations_id_seq OWNED BY public.github_install
 --
 
 CREATE TABLE public.github_pull_requests (
-    id bigint NOT NULL,
+    id integer NOT NULL,
```
Comment (gregberge) [inline apps/database/db/structure.sql:404]: Not good

**Review by gregberge — CHANGES_REQUESTED** [review]
Something weird in the structure but the rest is OK 👌. Maybe just check the type in production, report it in your db and in the structure. But I think it should be bigint.

**Pull request merged**